patches/tModLoader/Terraria/release_extras/DedicatedServerUtils/ManageServer.sh

```diff
@@ -0,0 +1,396 @@
+		do
+			steamcmdCommand="$steamcmdCommand +workshop_download_item 1281930 $line"
+		done
+		steamcmd "$steamcmdCommand +quit"
+	else
+		echo "No workshop mods to install"
+	fi
+}
+
+function install_mods {
+	script_path=$(realpath "$0") # Get path to the script, in case someone launched it from another directory
+	if [[ -v checkdir ]]
+	then
+		pushd "$checkdir"
+	else
+		pushd "$(dirname "$script_path")"
+	fi
+
+	if ! command -v steamcmd &> /dev/null
+	then
+		echo "steamcmd not found on path, no workshop mods will be installed or updated"
+	else
+		update_workshop_mods
+	fi
+
+	if [[ -v mods_path ]]
+	then
+		echo "Moving mods to $mods_path"
+	elif [[ -v XDG_DATA_HOME ]]
+	then
+		mods_path=$XDG_DATA_HOME/Terraria/tModLoader/Mods
+		echo "Moving mods to $mods_path"
+	else
+		mods_path=~/.local/share/Terraria/tModLoader/Mods
+		echo "Moving mods to $mods_path"
+	fi
+
+	mkdir -p $mods_path
+
+	# If someone has .tmod files this will install them
+	if [[ -f "*.tmod" ]]
+	then
+		echo "Copying .tmod files to the Mods directory"
+		cp ./*.tmod $mods_path
+	fi
+
+	# Move enabled.json to the right place
+	if [[ -f "enabled.json" ]]
+	then
+		echo "Copying enabled.json to the Mods directory"
+		cp enabled.json $mods_path
+	fi
+
+	popd
+}
+
+function print_version {
+	echo "tML Maintenance Tool: v$script_version"
+
+	if [[ -v folder ]]
+	then
+		echo tML Install: "$(cat "$folder/.ver")"
+	elif [[ -r ~/tModLoader/.ver ]]
+	then
+		echo tML Install: "$(cat ~/tModLoader/.ver)"
+	fi
+
+	exit 0
+}
+
+function print_help {
+	echo \
+"tML dedicated server installation and maintenance script
+
+Options:
+ -h|--help           Show command line help.
+ -v|--version        Display the current version of the tool and a tModLoader Github install.
+ -i|--install        Install tModLoader and mods.
+ -u|--update         Update an existing tModLoader installation and mods.
+ -g|--github         Use the binary off of Github instead of using steamcmd.
+ -f|--folder         Choose the folder to update/install to. When using steamcmd, make sure to use an absolute path. Default path is ~/tModLoader when using Github, and ~/Steam/steamapps/common/tModLoader when using steamcmd.
+ -m|--mods-path      The path to your mods folder. Any .tmod files and enabled.json are sent here.
+ --username          The steam username to login with. Only applies when using steamcmd.
+ --tml-version       The version of tML to download. Only applies when using Github. This should be the exact tag off of Github (ex. v2022.06.96.4).
+ --update-script     Update the script to the latest version on Github.
+ --mods-only         Only install/update mods.
+ --no-mods           Don't install/update mods.
+ --check-dir         Directory to check for enabled.json, install.txt, and any .tmod files.
+
+When running --install and --update, enabled.json, install.txt, and any .tmod files will be checked for in the location of the script or in the directory specified by --check-dir."
+	exit
+}
+
```

Comment:
Can I get a comment block indicating this is entry point of main script? 
Easy to get lost with functions listed first

Reply:
Done.